### ISSUE TITLE
pick_value tool at v0.2.0 misbehaves when first value is null

### ISSUE THREAD

**Issue by wm75** [galaxyproject/tools-iuc#6315]
Observed with a WF from @Sch-Da.

When the first value to pick from is the output of a skipped step that dataset's type is expression.json and with the change in #5716 this is what will be used as the output's datatype and hang at metadata detection afterwards (if the picked dataset's content is sufficiently complex I guess).

Version 0.1.0 of the tool works fine with the inputs in the WF (though requiring a change of datatype data to a fixed known datatype).

@Sch-Da maybe you can share a minimal example history here?

**Comment by Sch-Da** [comment]
Sure, thank you @wm75 
https://usegalaxy.eu/u/schnda/h/bug-report---minimal-example-pick-a-value
Here, the steps 10 and 12 run endlessly. I hope that helps. Otherwise, let me know, if you need more.

**Comment by Sch-Da** [comment]
Additionally, I have trouble setting the datatype to text in version 0.1.0. If I do this in the WF, like here:
https://usegalaxy.eu/u/schnda/h/data-type-set-to-text
I immediately get an error in the WF editor, so the workaround from @wm75 does not seem to apply here.

![screenshot 178](https://github.com/user-attachments/assets/e5f4c782-342c-4ec3-8744-7aaf285750b2)

**Comment by wm75** [comment]
that should be just a consequence of changing the tool version while it's connected in the WF. Tool params are not identical between the versions so Galaxy alerts you that connections got broken.

**Comment by wm75** [comment]
You don't want to switch the *input* to text, but change the output format.

**Comment by Sch-Da** [comment]
![screenshot 179](https://github.com/user-attachments/assets/e911fb01-ffcc-4b56-8131-59b3afc3cd4f)

Yes, that's what I meant, sorry. I changed that only today and then got an error again.

**Comment by Sch-Da** [comment]
Or where else can I check the output value?

**Comment by wm75** [comment]
> Or where else can I change the output value?

well, under output options.

but maybe we shouldn't mix this with the actual issue here.

@bernt-matthias do you want to try and reproduce the original problem? Otherwise I could take a closer look next week.

**Comment by mvdbeek** [comment]
> When the first value to pick from is the output of a skipped step that dataset's type is expression.json and with the change in https://github.com/galaxyproject/tools-iuc/pull/5716 this is what will be used as the output's datatype

I don't think this is a tool bug, a null value produced by a skip step should just be a null value for expression tools, and that's not happening correctly.
